Upstream-Status: Pending

# with gcc 4.3.3 and MIPS target, gcc "internal compiler error: segmentation fault" error
# occurs when compiling Applydeltarpm.cc. I didn't find the exact reason leading to this
# error, but it should be related to how boost tribool type is interpreted by g++. So provide
# this mips workaround for now. We'll come back later to check whether it's still required after
# upgrading to 4.5.0.
#
# Signed-off-by Kevin Tian <kevin.tian@intel.com>, 2010-09-08

diff --git a/zypp/repo/Applydeltarpm.cc b/zypp/repo/Applydeltarpm.cc
index 7b382be..0d7a1f8 100644
--- a/zypp/repo/Applydeltarpm.cc
+++ b/zypp/repo/Applydeltarpm.cc
@@ -77,7 +77,8 @@ namespace zypp
       else
         {
           // _last is 'indeterminate' or '!have'
-          if ( (_last = have) )
+          _last = have;
+          if ( _last )
             MIL << "Found executable " << prog << endl;
           else
             WAR << "No executable " << prog << endl;
